### ISSUE TITLE
Replace raw-input with input in Python 2

### ISSUE THREAD

**Issue by joehakimrahme** [hylang/hy#259]
Python2's `input` function is bad. If you want to `eval` the input, you should do so explicitly. This is why Python2 introduced `raw_input` and why Python3 made `input` behave like `raw_input`.

I think Hy should have a single function `input` that **doesn't** evaluate the string it gets. It should automatically translate into `raw_input` if translated to Python2 and `input` if translated to Python 3.

### Cons
I think this might affect the holy Python <-> Hy interoperability. 

### However
I think we do something similar with division in Hy, where `(/ 3 2)` always evaluates to `1.5` regardless of the version of Python it compiles to.

Basically, I'd like to get your opinion before starting coding this one. We could discuss the technical details of the implementation, but first let's focus on deciding if it's a sound design choice.

**Comment by Foxboron** [comment]
I believe this is fixed now!
